doc/design/show_train_stmt.md

```diff
@@ -0,0 +1,52 @@
+# Show Train Statement in SQLFlow
+
+## Background
+
+Like using other SQL engines, people may want to access the table metadata (say, the  table creating statement) using a SQL query. Usually they will type a statement like:
+```
+SHOW CREATE TABLE `test`;
+```
+which will produce:
+```
+mysql> SHOW CREATE TABLE `test`;
++-------+------------------------------------------------------------------+
+| Table | Create Table                                                                                                                                                                                                                                                          |
++-------+------------------------------------------------------------------+
+| test  | CREATE TABLE `test` (
+  `sepal_length` float DEFAULT NULL,
+  `sepal_width` float DEFAULT NULL,
+  `petal_length` float DEFAULT NULL,
+  `petal_width` float DEFAULT NULL,
+  `class` int DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
++-------+-----------------------------------------------------------------+
+```
+
+It worth for SQLFlow to do the same thing. On SQLFlow platform, user may create a lot of models with a minor set of parameters different with each other. After a while, the  user may hardly tell what the differences are if he hadn't record the information else where (like in model name).  We would like to provide a formal way to solve this problem. Here we introduce the SHOW TRAIN feature.
+
+## Grammar
+```
+SHOW TRAIN table_where_model_stored;
+```
+where
+- `table_where_model_stored` must be a table which stores the trained model in MySQL or Hive
+
+This query will show the original train sql statement like:
+```
++-------+------------------------------------------------------------------+
+| Table | Train Statement
++-------+------------------------------------------------------------------+
+| test  | SELECT * FROM train to TRAIN xgboost.gbtree with learning_rate=0.4,
+ objective=multi:softmax, num_class=3 LABEL class INTO test;
++-------+-----------------------------------------------------------------+
+```
+In the future we may support to show more metadata other than the original sql statement.
+
+## Implementation
+
+- Extend the SQLFlow parser with our `SHOW TRAIN` statement
```

Comment:
How? The current parser implementation will call the dialect parser and getting an error position and parse the extended train statements. To parse `SHOW TRAIN ...` we may need to parse if the statement is `SHOW TRAIN ...` before calling the dialect parser.

Reply:
Yes, as with this case, maybe we can do some hard-coded preprocessing to escape the dialect parser.
However, as we introduce more and more grammer features, we have to distinguish each kind of stmt sooner or later. Not all case can fit the current `SELECT ... TO ...` pattern. We may need a more powerful customized parser to extract standard SQL and the extended SQL in a unified manner.

Reply:
Maybe mentioning some implementation details on "a more powerful customized parser".